megamek/src/megamek/client/ui/swing/boardview/BoardView1.java

```diff
@@ -569,57 +569,31 @@ public void mouseWheelMoved(MouseWheelEvent we) {
                 double ihdx = ((double)inhexDelta.x)/((double)HEX_W)/scale;
                 double ihdy = ((double)inhexDelta.y)/((double)HEX_H)/scale;
                 int oldzoomIndex = zoomIndex;
+                
+                boolean ZoomNoCtrl = GUIPreferences.getInstance().getMouseWheelZoom();
+                boolean wheelFlip = GUIPreferences.getInstance().getMouseWheelZoomFlip();
+                boolean zoomIn = (we.getWheelRotation() > 0) ^ wheelFlip; // = XOR
+                boolean doZoom = ZoomNoCtrl ^ we.isControlDown(); // = XOR
+                boolean horizontalScroll = !doZoom && we.isShiftDown();
```

Comment:
While in testing this does what it says it does, the use of XOR could likely use a small comment/table explaining which modes are active with which key presses. Not required, but would be helpful in the future.